src/ATen/DLConvertor.cpp

```diff
@@ -125,28 +125,43 @@ static ScalarType getATenScalarType(const DLDataType& dtype) {
 }
 
 
-// This function returns a shared_ptr to DLpack tensor constructed out ATen tensor
-DLTensor* toDLPack(const Tensor& src, DLTensor* dlTensor) {
-  dlTensor->data = src.data_ptr();
+void destructor(void * arg) {
```

Comment:
I think this should be:
```
static void destructor(DLManagedTensor * self) {
    delete  static_cast<ATenDLMTensor*>(self->ctx);
}
```

Reply:
why not just pass the ctx directly? when we call  destructor, we can pass the context as well directly

Reply:
Let's say my DLTensor was allocated like so:

```
void deleter(DLManagedTensor * self) {
   delete self->tensor.data;
   delete self->tensor.sizes;
   delete self->tensor.strides;
}
....
DLManagedTensor t;
t.tensor.data = new float[32];
t.tensor.sizes = new int[2];
t.tensor.strides = new int[32];
t.ctx = nullptr; //unused
t.deleter = deleter;
...
```

In this case, it is very convenient to have a pointer to `self` in the deleter.

Reply:
makes sense! I'll update the PR. Thanks for catching and explaining! :)

Reply:
are we fine having struct `DLManagedTensor {}` in dlpack.h rather than `typedef struct {} DLManagedTensor`. The former is what's been followed in rest of header file